src/persistence/stock.rs

```diff
@@ -741,32 +740,21 @@ impl<S: StashProvider, H: StateProvider, P: IndexProvider> Stock<S, H, P> {
 
         let genesis = self.stash.genesis(contract_id)?.clone();
         // get schema supplyment by schema id
-        match self
-            .stash
+        self.stash
             .supplement(ContentRef::Schema(genesis.schema_id))?
-        {
-            Some(schema_suppl) => supplyments.insert(schema_suppl.clone()),
-            None => false,
-        };
+            .and_then(|schema_suppl| Some(supplements.insert(schema_suppl.clone())));
 
         let schema_ifaces = self.stash.schema(genesis.schema_id)?.clone();
         let mut ifaces = BTreeMap::new();
         for (iface_id, iimpl) in schema_ifaces.iimpls {
             let iface = self.stash.iface(iface_id)?;
             // get iface and iimpl supplyment by iface id and iimpl id
-            match (
-                self.stash.supplement(ContentRef::Iface(iface.iface_id()))?,
-                self.stash
-                    .supplement(ContentRef::IfaceImpl(iimpl.impl_id()))?,
-            ) {
-                (Some(iface_suppl), Some(iimpl_suppl)) => {
-                    supplyments.insert(iface_suppl.clone());
-                    supplyments.insert(iimpl_suppl.clone())
-                }
-                (Some(iface_suppl), None) => supplyments.insert(iface_suppl.clone()),
-                (None, Some(iimpl_suppl)) => supplyments.insert(iimpl_suppl.clone()),
-                (None, None) => false,
-            };
+            self.stash
+                .supplement(ContentRef::Iface(iface.iface_id()))?
+                .and_then(|iface_suppl| Some(supplements.insert(iface_suppl.clone())));
```

Comment:
Here and avove, actually, this should be `map`:

```suggestion
                .map(|iface_suppl| supplements.insert(iface_suppl.clone()));
```